### ISSUE TITLE
Support for WEBSITE_TIME_ZONE in Linux consumption

### ISSUE THREAD

**Issue by balag0** [Azure/azure-functions-host#5246]

**Comment by paxos** [comment]
This needs to have a higher priority. The provided workaround, which is scheduling everything in UTC, doesn't work for all scenarios - or I am missing an obvious solution for this.

or example, there is no way to have the function run every 3 hours between 9am-6pm every work day. The translated UTC schedule would be:

`0 0 16,19,22,1 * * 1,2,3,4,5`

But this will run once Sunday, and miss one run on Friday:
```
5/17/2020 6:00:00 PM
5/18/2020 9:00:00 AM
5/18/2020 12:00:00 PM
5/18/2020 3:00:00 PM
5/18/2020 6:00:00 PM
5/19/2020 9:00:00 AM
5/19/2020 12:00:00 PM
5/19/2020 3:00:00 PM
5/19/2020 6:00:00 PM
5/20/2020 9:00:00 AM
5/20/2020 12:00:00 PM
5/20/2020 3:00:00 PM
5/20/2020 6:00:00 PM
5/21/2020 9:00:00 AM
5/21/2020 12:00:00 PM
5/21/2020 3:00:00 PM
5/21/2020 6:00:00 PM
5/22/2020 9:00:00 AM
5/22/2020 12:00:00 PM
5/22/2020 3:00:00 PM
```

**Comment by krish-gh** [comment]
@balag0 Is it only for Consumption plan? I see user reported it for Linux Premium or App Service plan also. https://stackoverflow.com/questions/64316818/time-zone-issue-python-function-app-on-azure

**Comment by balag0** [comment]
Yes, this issue is specific to consumption plans only.

**Comment by rodchenkov** [comment]
Hi, any updates on it? Looks like I just got this issue.

**Comment by NvHeijningen** [comment]
Hi, got the same problem here. Cant find a solution. Any updates?

**Comment by balag0** [comment]
@NvHeijningen @rodchenkov yes, this is still a limitation and we don't have an ETA yet. Are you able to workaround this in the meantime?

**Comment by pedrojunqueira** [comment]
Some issue here. Set up `WEBSITE_TIME_ZONE` configuration and still running on UTC `tz`

**Comment by jkdmyrs** [comment]
@balag0 this has been open since 2019 with no suggested work arounds. @paxos gave an example of which there seems to be no work around. Will this ever be fixed? Please advise.

**Comment by balag0** [comment]
Thanks for the ping.. I have opened an internal WI for prioritization. Will update here once we have a plan.

In the meantime, one possible (hacky) workaround, set the frequency greater than required timer frequency. Write code in the function code to check time and either do work or silently exit based on the current time in required timezone.

Ex: To run every 3 hours between 9am-6pm every work day

Configure the trigger to fire once an hour every day. In function code, get current time in utc, check if it is time to run the function , then either silently return or run the actual function logic. Understand its not ideal, but this might help unblock while the feature is coming online.

**Comment by TroyWitthoeft** [comment]
Would love some improvement here, please! I know we've documented the deficiency [here](https://docs.microsoft.com/en-us/azure/azure-functions/functions-bindings-timer?tabs=in-process&pivots=programming-language-python#ncrontab-time-zones), thanks. But this issue still has the opportunity to surprise teams.  For instance, say we use consumption in a lower environments and app service in production environments.  Using Windows, all is well.  Using Linux, then :tada: ! 

We've got the workaround suggestion from @balag0, but - as mentioned - it's not ideal.  It passes a bit of code complexity onto the customers to solve a product inconsistency.   Hopefully this gets picked up to work on?  Anywhere we can help?  Happy to contribute.  Not sure where to start, but currently ruminating over [this portion](https://github.com/Azure/azure-functions-host/blob/f1c6016b8e3a0f6ad8f02eb04297badf118e7acb/src/WebJobs.Script.WebHost/Standby/StandbyManager.cs#L97) of the azure functions host code ...

If it's too far back on the backlog, could we consider some sort of mitigation?  Say ... have the host detect when the app is Linux Consumption and the `WEBSITE_TIME_ZONE` configuration value is set and warn the user?  Popup a message in the portal?

**Comment by balag0** [comment]
@TroyWitthoeft This is a great point. Making a change to display a warning in "Diagnose and solve problems" -> "Configuration and Management" section in the portal if the app is using "WEBSITE_TIME_ZONE " or "TZ" appsettings.

**Comment by Vox1984** [comment]
+1 for proper timezones in 2023

**Comment by alicelupsan** [comment]
+1 for proper timezones in 2023

**Comment by drk-mtr** [comment]
The suggestion to display a warning in "Diagnose and solve problems" until this is resolved is a good one. I have only just stumbled across the documentation that says "this isn't supported" after 2 days (on and off) of trying to solve this.

**Comment by daniel17903** [comment]
+1 for proper timezones in 2024

**Comment by Ben-tal** [comment]
+1 for proper timezones in 2025

**Comment by Tomasz-Kluczkowski** [comment]
yes please
